### ISSUE TITLE
MMS Group Messaging

### ISSUE THREAD

**Issue by jgoulet1994** [andyholmes/gnome-shell-extension-gsconnect#553]
**Is your feature request related to a problem? Please describe.**
It would be convenient for gsconnect to include group messages.

**Describe the solution you'd like**
The messages window shows group messages.

**Describe alternatives you've considered**
N/A

**System Details (please complete the following information):**
 - GSConnect version: 24
   - Installed from: extensions.gnome.org

**Comment by andyholmes** [comment]
I assume you mean group messages sent over MMS? Currently kdeconnect-android doesn't support MMS messages.

**Comment by jgoulet1994** [comment]
That is what I meant, thanks for the quick response.

**Comment by andyholmes** [comment]
We'll have to wait for MMS support upstream before that's possible. Unfortunately kdeconnect-kde still doesn't ship it's SMS app, so that might be awhile :)

**Comment by qfire** [comment]
Just an FYI, kdeconnect-android now has basic MMS support now, see [merge request](https://invent.kde.org/kde/kdeconnect-android/merge_requests/78).

Is anyone already working on adding this to GSConnect?

I'll look at it myself if I can find some time, but I suspect someone else will get to it before I do.

**Comment by ferdnyc** [comment]
Caveats being:
* The MR for their _desktop_ app [is still open](https://invent.kde.org/kde/kdeconnect-kde/merge_requests/97), so this is currently unusable even within the KDE ecosystem.
* Because of that, it sounds like it's completely untested
* I'm not sure what it really buys us at this stage of development, since (quoting from the android MR)
    - > This patch adds basic MMS support for plain-text MMS, including multi-target messages.
    - > Multi-target messages do not have the full list of recipients (I am planning to work on this in another patch, because this one is already quite large enough)

So, it's still text-only, and supports multiple recipients... but multiple recipients support is completely broken. **#Ummmm?**

**Comment by andyholmes** [comment]
Basically what that MR adds is pulling text-only MMS messages from the database and sending them to be handled as regular SMS messages. These should already "just work", mostly. @sredman's fork has a [mms-multitarget](https://invent.kde.org/sredman/kdeconnect-android/tree/mms-multitarget) branch that's still a work in progress.

There's still discussion happening about how to best handle group messages, since sending and receiving MMS messages requires changes to the packet protocol. MMS attachments will be a whole other thing.

**Comment by ferdnyc** [comment]
However, the MR comes with free froyo!

The froyo is also cursed.

**Comment by sredman** [comment]
> Caveats being:
> 
>     * The MR for their _desktop_ app [is still open](https://invent.kde.org/kde/kdeconnect-kde/merge_requests/97), so this is currently unusable even within the KDE ecosystem.

The whole of the KDE Connect SMS is unreleased, so at some level the whole SMS plugin is unusable... I think I am the only one using it, since when I have accidentally broken the version on master nobody has complained and I have noticed weeks later

The lastest desktop MR is this, btw: https://invent.kde.org/kde/kdeconnect-kde/merge_requests/101 
> 
>     * Because of that, it sounds like it's completely untested
>

I use it on the daily so it's mostly tested. At least, it works for the things I do with it

> Basically what that MR adds is pulling text-only MMS messages from the database and sending them to be handled as regular SMS messages. These should already "just work", mostly. @sredman's fork has a [mms-multitarget](https://invent.kde.org/sredman/kdeconnect-android/tree/mms-multitarget) branch that's still a work in progress.
> 

I'm ready to merge that branch (https://invent.kde.org/kde/kdeconnect-android/merge_requests/80). At some level, I'm waiting for you to say you're ready with the support for the changes to the message packet since I don't want to throw you under the bus with a surprise release :)

> However, the MR comes with free froyo!
> 
> The froyo is also cursed.

I would more say that "it doesn't not come with Froyo" -- Everything I have read suggests that maybe it should work down to versions as old as that, but I don't have any live Android devices which send/receive SMS to test with. My old phone that I was using for testing these things seems to have broken :(

**Comment by ferdnyc** [comment]
> > However, the MR comes with free froyo!
> > The froyo is also cursed.
> 
> I would more say that "it doesn't not come with Froyo" -- Everything I have read suggests that maybe it should work down to versions as old as that, but I don't have any live Android devices which send/receive SMS to test with. My old phone that I was using for testing these things seems to have broken :(

For the record, I was _attempting_ to reference [this Simpsons bit (YouTube link)](https://youtu.be/YtmTYTWWhOA?t=25), but forgot that:

1. "Froyo" was also the name of an Android release
1. The old man calls it "Frogurt"

Put together, those two errors turn my ham-handed attempt at humor into an utter disaster.

**Comment by andyholmes** [comment]
With #585 being merged, GSConnect now has basic support for viewing group MMS messages, but code for sending messages has not been merged into kdeconnect-android yet. A lot of the messaging code had to be rewritten, so testing will be necessary.

**Comment by andyholmes** [comment]
Closing since we are now waiting on support in `kdeconnect-android` for any more features.
